datafusion/physical-optimizer/src/filter_pushdown.rs

```diff
@@ -597,3 +631,34 @@ fn push_down_filters(
     }
     Ok(res)
 }
+
+fn allow_pushdown_for_expr(expr: &Arc<dyn PhysicalExpr>) -> bool {
+    let mut allow_pushdown = true;
+    expr.apply(|e| {
+        allow_pushdown = allow_pushdown && allow_pushdown_for_expr_inner(e);
+        if allow_pushdown {
+            Ok(TreeNodeRecursion::Continue)
+        } else {
+            Ok(TreeNodeRecursion::Stop)
+        }
+    })
+    .expect("Infallible traversal of PhysicalExpr tree failed");
+    allow_pushdown
+}
+
+fn allow_pushdown_for_expr_inner(expr: &Arc<dyn PhysicalExpr>) -> bool {
+    // TODO: do downcast matching on specific expressions, or add a method to the PhysicalExpr trait
+    // to check if the expression is volatile or otherwise unsuitable for pushdown.
+    if let Some(scalar_function) =
+        expr.as_any()
+            .downcast_ref::<datafusion_physical_expr::ScalarFunctionExpr>()
+    {
+        let name = scalar_function.fun().name();
```

Comment:
I think you can find volatility using

https://docs.rs/datafusion/latest/datafusion/logical_expr/trait.ScalarUDFImpl.html#tymethod.signature

Which then has a volatility field: https://docs.rs/datafusion/latest/datafusion/logical_expr/struct.Signature.html#structfield.volatility